### DISCUSSION TITLE
Building wheel for h5py failing due to non-empty directory

### DISCUSSION THREAD

**Discussion by lindsayad** [firedrakeproject/firedrake#3713] [Firedrake support]
Hi I'm building firedrake on a new HPC machine and hitting this error:
```
  removing build/bdist.linux-x86_64/wheel
  error: [Errno 39] Directory not empty: 'build/bdist.linux-x86_64/wheel/h5py'
  error: subprocess-exited-with-error
  
  × Building wheel for h5py (pyproject.toml) did not run successfully.
```
If I go into that directory, it's indeed not empty
```
lindad@bitterroot2:/scratch/lindad/bitterroot-firedrake/src/h5py/build/bdist.linux-x86_64/wheel/h5py$ ls -lrth
total 512
d-w------- 2 lindad lindad 4.0K Jul 17 17:46 tests
```
Any suggestions? I'm guessing this is an h5py issue so will be happy to report upstream

[firedrake-install.log](https://github.com/user-attachments/files/16417056/firedrake-install.log)

**Comment by connorjward** [comment]
I do not know why you are hitting this. I can't see anything wrong from the log that you have shared and googling doesn't really yield any useful information.

The specific command that is failing (with the venv active) is:
```
$ python -m pip install --no-build-isolation --no-binary mpi4py,randomgen,numpy -v --no-build-isolation $VIRTUAL_ENV/src/h5py/
```

That might be a good place to start from in terms of producing a minimal failing example or maybe try upgrading/downgrading pip, Cython, setuptools etc.

Apologies I can't be of more help.

**Reply by lindsayad** [reply]
No worries. I appreciate the quicky reply. I've opened https://github.com/h5py/h5py/issues/2467. Maybe somebody there will also have a suggestion
